packages/ERTP/src/extentOps/helpers.js

```diff
@@ -0,0 +1,65 @@
+import harden from '@agoric/harden';
+import { passStyleOf } from '@agoric/marshal';
+import { insist } from '@agoric/insist';
+
+const makeObjListExtentOps = (customInsistFn, getIdentifierFn) => {
+  const identity = harden([]);
+  const extentOps = harden({
+    insistKind: list => {
+      insist(passStyleOf(list) === 'copyArray')`list must be an array`;
+      list.forEach(customInsistFn);
+    },
+    empty: _ => identity,
+    isEmpty: list => passStyleOf(list) === 'copyArray' && list.length === 0,
+    includes: (left, right) => {
+      const leftIds = left.map(getIdentifierFn);
+      const rightIds = right.map(getIdentifierFn);
+      const leftWeakSet = new WeakSet(leftIds);
+      const has = elem => leftWeakSet.has(elem);
+      return rightIds.every(has);
+    },
+    equals: (left, right) =>
+      extentOps.includes(left, right) && extentOps.includes(right, left),
+    with: (left, right) => {
+      const unionWeakSet = new WeakSet();
+      const result = [];
+      left.forEach(extent => {
```

Comment:
Why add each element separately for both lists? Wouldn't it be clearer to add all the elements from both to a Set, and then add all the members of the Set to result?

Reply:
Since it's a WeakSet, we can't get anything from the WeakSet